runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java

```diff
@@ -258,4 +333,107 @@ public void onRemovedFriend(RemovedFriend event)
 		log.debug("Remove friend: '{}'", displayName);
 		setFriendNote(displayName, null);
 	}
+
+	@Subscribe
+	public void onScriptCallbackEvent(ScriptCallbackEvent event)
+	{
+		if (!config.showIcons())
+		{
+			return;
+		}
+
+		switch (event.getEventName())
+		{
+			case "friend_cc_settext":
+			case "ignore_cc_settext":
+				String[] stringStack = client.getStringStack();
+				int stringStackSize = client.getStringStackSize();
+				final String rsn = stringStack[stringStackSize - 1];
+				final String sanitized = Text.toJagexName(Text.removeTags(rsn));
+				currentlyLayouting = sanitized;
+				if (getFriendNote(sanitized) != null)
+				{
+					stringStack[stringStackSize - 1] = rsn + " <img=" + iconIdx + ">";
```

Comment:
I would maybe also check iconidx somehow isn't set here

Reply:
not sure if thats what you meant but i added a guard clause if iconidx == -1 return